### PR TITLE
Add default case for RAJA_XXX macros

### PR DESCRIPTION
**Pull request by davidbeckingsale**
This emits a warning if the RAJA_COMPILER is unknown, and defines the RAJA_INLINE, RAJA_SIMD and RAJA_DATA_ALGIN(d) macros as nothing.

### CODE REVIEW

**Comment by trws** [comment]
For RAJA_INLINE and RAJA_ALIGN_DATA is there a reason not to use the C++11 standard keyword/attribute in the default version?

**Comment by davidbeckingsale** [comment]
Probably not, good idea.

**Comment by rhornung67** [comment]
No objection if it works with all compilers we need it to.  Many of the band-aids and bits of chewing gum you see in the code are a result of poor compiler support in the past.

**Comment by trws** [comment]
Good point @rhornung67.  For the align macro, gcc is one where the default wouldn't work (as well as we would like) so it's overridden.  Are there compilers we explicitly care about that don't already have a block to set those macros explicitly?

**Comment by davidbeckingsale** [comment]
@trws what's the standard C++11 equivalent of the `assume_aligned` stuff?

**Comment by trws** [comment]
Crud, I was thinking of the other alignment macro that hasn't merged yet, assume_aligned is non-portable. I was thinking of the command to align something, the `[[aligned]]` attribute.

File: include/RAJA/config.hxx.in [hunk 1]
```diff
@@ -224,6 +224,14 @@ const int COHERENCE_BLOCK_SIZE = @RAJA_COHERENCE_BLOCK_SIZE@;
 
 #define RAJA_SIMD  // TODO: Define this...
 
+#else
+
+#pragma message("RAJA_COMPILER unknown, using default empty macros.")
+
+#define RAJA_INLINE inline
```
Comment (davidbeckingsale) [inline include/RAJA/config.hxx.in]: I think we agreed in the dev meeting that `inline` was the appropriate option for the `RAJA_INLINE` macro.

**Comment by davidbeckingsale** [comment]
This should be ready for merging once the Travis builds are done

**Comment by trws** [comment]
Approved, out of date wrt base branch, so I'm going to hit update and make sure the build passes with that before merging.

**Comment by davidbeckingsale** [comment]
Sounds good, go ahead and merge when you are happy.

**Pull request merged**